lib/immutable.js

```diff
@@ -109,7 +109,7 @@ Immutable = {
         }
         if (typeHandler.type.toLowerCase() === 'array' && Array.isArray(data)) {
           return typeHandler.handler(data, builder, options);
-        } else if (typeHandler.type.toLowerCase() === 'objectid' && data._bsontype === 'ObjectID') {
+        } else if (typeHandler.type.toLowerCase() === 'objectid' && data._bsontype === 'ObjectId') {
```

Comment:
so now that we are super consistent about using ObjectId - does this actually need toLower?